cmd/mimir/help-all.txt.tmpl

```diff
@@ -1145,6 +1145,12 @@ Usage of ./cmd/mimir/mimir:
     	Per-tenant allowed push request burst size. 0 to disable.
   -distributor.request-rate-limit float
     	Per-tenant push request rate limit in requests per second. 0 to disable.
+  -distributor.retry.base duration
+    	[experimental] Base time to wait before retrying a 429/5xx request. (default 3s)
+  -distributor.retry.enabled
+    	[experimental] Enabled controls inclusion of the Retry-After header in the response: true includes it for client retry guidance, false omits it.
```

Comment:
instead of having another flag, can we have the value of `0` mean that the header is disabled? Or does `0` have a special meaning in `Retry-After`?

Reply:
normally agent would ignore the retry-after if it is lower than their retry interval calculated. But to be sure about the behavior i would prefer to not set retry-after at all when disabled, so it is clear agent could fallback always to the default setting https://github.com/grafana/agent/blob/70f8e9575f63883daacfb84cf33da0820c0f0c2a/pkg/config/agentmanagement.go#L240

Reply:
I agree we should not set it at all, if it's disabled on mimir side.

Reply:
are we writing this feature for the agent? Aren't there some more established conventions around this flag? I don't think we should be coupling with any one client in particular

Reply:
What's the point of setting Retry-After to 0? "Please don't wait?" That would be weird thing to send back, and I don't think we should do that. Futhermore this value should be integer, so minimum positive value is 1sec.

Reply:
For the moment, We will not put retry-after header when it is disabled, only set it when enabled.